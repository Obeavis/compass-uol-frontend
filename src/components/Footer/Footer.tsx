import { FC, HTMLAttributes } from "react";

interface Props extends HTMLAttributes<any> {
  countdown: number;
  reset: Function;
  logout: Function;
}

const Footer: FC<Props> = ({ countdown, reset, logout }) => {
  return (
    <footer className="flex fixed items-center justify-center w-full bg-gradient-to-tr from-tertiary-dark to-tertiary bottom-0 lg:h-20 z-50">
      <div className="flex w-full items-center h-full flex-col lg:flex-row">
        <div className="flex lg:basis-1/2 justify-end pt-5 lg:pt-0 px-4 lg:px-10">
          <div className="flex lg:w-4/6">
            <span className="text-gray-200 text-xs lg:text-right">
              Essa janela do navegador é usada para manter sua sessão de
              autenticação ativa. Deixe-a aberta em segundo plano e abra uma
              nova janela para continuar a navegar.
            </span>
          </div>
        </div>
        <div className="hidden lg:flex border-r border-gray-100 h-16"></div>
        <div className="flex flex-col lg:flex-row lg:basis-1/2 justify-between items-center h-full w-full pt-5 lg:pt-0 lg:pl-20">
          <div className="flex items-center">
            <span className="text-gray-200 text-xs w-24">
              Application refresh in
            </span>
            <div className="flex flex-col items-center w-24">
              <span className="text-gray-200 text-3xl font-bold">
                {countdown}
              </span>
              <span className="text-gray-200 text-xs">seconds</span>
            </div>
          </div>
          <div className="flex items-center pt-5 lg:pt-0 w-full lg:w-auto h-20 lg:h-full">
            <div
              className="flex basis-1/2 justify-center items-center bg-white h-full cursor-pointer lg:w-32"
              onClick={() => reset()}
            >
              <span className="text-quartiary text-sm">
                Continuar Navegando
              </span>
            </div>
            <div
              className="basis-1/2 flex h-full items-center justify-center cursor-pointer"
              onClick={() => logout()}
            >
              <span className="text-gray-100 text-sm px-6">Logout</span>
            </div>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
