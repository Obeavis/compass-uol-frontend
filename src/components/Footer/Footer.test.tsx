import { render } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer component", () => {
  test("renders the component", async () => {
    const reset = jest.fn();
    const logout = jest.fn();
    const { container } = render(
      <Footer logout={logout} reset={reset} countdown={100} />
    );

    expect(container).toMatchSnapshot();
  });
});
