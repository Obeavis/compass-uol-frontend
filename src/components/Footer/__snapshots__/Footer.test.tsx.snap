// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the component 1`] = `
<div>
  <footer
    class="flex fixed items-center justify-center w-full bg-gradient-to-tr from-tertiary-dark to-tertiary bottom-0 lg:h-20 z-50"
  >
    <div
      class="flex w-full items-center h-full flex-col lg:flex-row"
    >
      <div
        class="flex lg:basis-1/2 justify-end pt-5 lg:pt-0 px-4 lg:px-10"
      >
        <div
          class="flex lg:w-4/6"
        >
          <span
            class="text-gray-200 text-xs lg:text-right"
          >
            Essa janela do navegador é usada para manter sua sessão de autenticação ativa. Deixe-a aberta em segundo plano e abra uma nova janela para continuar a navegar.
          </span>
        </div>
      </div>
      <div
        class="hidden lg:flex border-r border-gray-100 h-16"
      />
      <div
        class="flex flex-col lg:flex-row lg:basis-1/2 justify-between items-center h-full w-full pt-5 lg:pt-0 lg:pl-20"
      >
        <div
          class="flex items-center"
        >
          <span
            class="text-gray-200 text-xs w-24"
          >
            Application refresh in
          </span>
          <div
            class="flex flex-col items-center w-24"
          >
            <span
              class="text-gray-200 text-3xl font-bold"
            >
              100
            </span>
            <span
              class="text-gray-200 text-xs"
            >
              seconds
            </span>
          </div>
        </div>
        <div
          class="flex items-center pt-5 lg:pt-0 w-full lg:w-auto h-20 lg:h-full"
        >
          <div
            class="flex basis-1/2 justify-center items-center bg-white h-full cursor-pointer lg:w-32"
          >
            <span
              class="text-quartiary text-sm"
            >
              Continuar Navegando
            </span>
          </div>
          <div
            class="basis-1/2 flex h-full items-center justify-center cursor-pointer"
          >
            <span
              class="text-gray-100 text-sm px-6"
            >
              Logout
            </span>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
